Platforms/QemuQ35Pkg/ConfigKnobs/ConfigDataGfx/ConfigDataGfx.c

```diff
@@ -64,16 +64,23 @@ ApplyGfxConfigToPolicy (
   GfxEnablePort0   = *(BOOLEAN *)ConfigBuffer;
   GfxSiliconPolicy = AllocateCopyPool (sizeof (DefaultQemuGfxPolicy), DefaultQemuGfxPolicy);
 
+  if (GfxSiliconPolicy == NULL) {
+    DEBUG ((DEBUG_ERROR, "Failed to allocate Policy structure\n"));
+    goto Exit;
+  }  // Mu
+ 
   // We only translate the GFX ports #0 exposed to platform from conf data
   GfxSiliconPolicy[0].Power_State_Port = GfxEnablePort0;
 
   Status = SetPolicy (&gPolicyDataGFXGuid, POLICY_ATTRIBUTE_FINALIZED, GfxSiliconPolicy, sizeof (DefaultQemuGfxPolicy));
+
   if (EFI_ERROR (Status)) {
     DEBUG ((DEBUG_ERROR, "%a Failed to update GFX policy per configuration data - %r!!!\n", __func__, Status));
     ASSERT (FALSE);
+    FreePool (GfxSiliconPolicy);   //Mu
     goto Exit;
   }
-
+  FreePool (GfxSiliconPolicy);   //Mu

```

Comment:
Since all of these ultimately free the `GfxSiliconPolicy` buffer and all execution flows through the `Exit` label on return, I'd consider initialize `GfxSiliconPolicy` to `NULL` and then add a single free if the pointer is non-null in the `Exit` path.

```c
Exit:
  if (GfxSiliconPolicy != NULL) {
    FreePool (GfxSiliconPolicy);
  }
  
  return Status;
```

Reply:
Good suggestion.  Updated w/ recommendations.